code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/TokenResourceManager.java

```diff
@@ -0,0 +1,92 @@
+package io.cattle.platform.iaas.api.auth.identity;
+
+import io.cattle.platform.iaas.api.auth.SecurityConstants;
+import io.cattle.platform.iaas.api.auth.integration.github.GithubConstants;
+import io.cattle.platform.iaas.api.auth.integration.interfaces.TokenCreator;
+import io.cattle.platform.iaas.api.auth.TokenUtils;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.factory.SchemaFactory;
+import io.github.ibuildthecloud.gdapi.model.ListOptions;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.impl.AbstractNoOpResourceManager;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+
+
+public class TokenResourceManager extends AbstractNoOpResourceManager {
+
+    private List<TokenCreator> tokenCreators;
+
+    public List<TokenCreator> getTokenCreators() {
+        return tokenCreators;
+    }
+
+    @Inject
+    public void setTokenCreators(List<TokenCreator> tokenCreators) {
+        this.tokenCreators = tokenCreators;
+    }
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[]{Token.class};
+    }
+
+    @Override
+    protected Object createInternal(String type, ApiRequest request) {
+        if (!StringUtils.equals(TokenUtils.TOKEN, request.getType())) {
+            return null;
+        }
+        try {
+            return createToken(request);
+        } catch (IOException e) {
```

Comment:
Nitpick: Its kind of silly that because of an implementation detail of GithubTokenCreator, this checked exception had to bubble all the way up here where it is then just converted to a runtime exception. Would have been better to convert it down in the the GithubTokenCreator and let it bubble up as an unchecked exception.

LdapTokenCreator doesnt actually even throw an IOException